### PR TITLE
Make location title and description editable.

### LINKED ISSUE #1175: Edit location name and description 🌍
## Description
A user should be able to edit the name and description of a location after it's been created.

## Relevant Job Stories
None

## Prerequisites
Don't start this until #1172 is done.

## Requirements
- [x] In the card that opens when a location is selected (selection implemented in #1172), it should be possible to edit the title and description in place
- [x] When a field is blurred, save the change to the API

## Possible implementations
Update the location by patching `/orgs/ID/locations/ID`.

Use `ZUIEditTextInPlace`?

### Location selected
Ignore the "Other events in this location" section which is tracked elsewhere.
image

### Editing title
image

### Editing description
image

## Open questions
* The design does not look like `ZUIEditTextInPlace`. Should we use another component, or change the design to be consistent with ZUIEditTextInPlace?

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR adds editing functionality to the title and description in the `LocationDetailsCard`. It also adresses a few comments from my earlier PR. It also says if there is no description, so the user can add one.

## Screenshots
bild

bild

bild

## Changes
* Adds methods on the `LocationsModel` to edit title and description.
* Uses `ZUIPreviewableInput` to render title and description on the card.
* Fixes some small things from earlier PR, like variable name and redundant state.

## Related issues
Resolves #1175

### CODE REVIEW

File: src/features/events/components/LocationModal/Map.tsx [hunk 1]
```diff
@@ -68,10 +66,13 @@ const Map: FC<MapProps> = ({
       }
       style={{ height: '80vh', width: '100%' }}
     >
-      <MapProvider>
+      <MapWrapper>
```
Comment (richardolsson) [inline src/features/events/components/LocationModal/Map.tsx:69]: Nice!

File: src/features/events/components/LocationModal/Map.tsx [hunk 2]
```diff
@@ -68,10 +66,13 @@ const Map: FC<MapProps> = ({
...
         {(map) => {
-          if (focusedMarker) {
-            map.setView(focusedMarker, 17);
+          if (selectedLocation) {
```
Comment (richardolsson) [inline src/features/events/components/LocationModal/Map.tsx:71]: Nice!

File: src/features/events/components/LocationModal/LocationDetailsCard.tsx [hunk 3]
```diff
@@ -1,23 +1,59 @@
...
   location,
 }) => {
   const messages = useMessages(messageIds);
+  const [title, setTitle] = useState(location.title);
+  const [description, setDescription] = useState(location.info_text);
+  const [fieldEditing, setFieldEditing] = useState<
+    'title' | 'description' | null
+  >(null);
+
+  const handleDescriptionTextAreaRef = useCallback(
```
Comment (richardolsson) [inline src/features/events/components/LocationModal/LocationDetailsCard.tsx:39]: Why is `ZUIPreviewableInput` not already handling this?

**Review by richardolsson — APPROVED** [review]
This looks great to me! I had a question about the use of a ref and "manual" focusing below, but I'm fine with merging as is.

File: src/features/events/components/LocationModal/LocationDetailsCard.tsx (hunk 3 above)
Comment (ziggabyte) [reply src/features/events/components/LocationModal/LocationDetailsCard.tsx:39]: For some reason I do not comprehend, it set focus at the start of the input when it is in multiline mode, so i just stole this from your work on the notes in views.

**Pull request merged**